dataRetriever/resolvers/headerResolver.go

```diff
@@ -108,9 +108,20 @@ func (hdrRes *HeaderResolver) resolveHeaderFromNonce(key []byte) ([]byte, error)
 	}
 
 	//Step 2. search the nonce-key pair first in datapool, second in storage
-	value, _ := hdrRes.hdrNonces.Get(nonce)
-	hash, ok := value.([]byte)
-	if hash == nil || !ok {
+	value, ok := hdrRes.hdrNonces.Get(nonce)
+	var hash []byte
+	if ok {
+		value.Range(func(shardId uint32, existingHash []byte) bool {
```

Comment:
is there no better way, than doing range over and over again ?

Reply:
Currently no, the sync.Map impl does not provide other functionality. Will think about having another implementation of the dataRetriever.ShardIdHashMap interface that could use better concurrent safe cachers.